JustSaying/AwsTools/MessageHandling/SqsNotificationListener.cs

```diff
@@ -216,29 +245,31 @@ private async Task<ReceiveMessageResponse> GetMessagesFromSqsQueue(string queueN
             }
         }
 
-        private async Task<int> GetNumberOfMessagesToReadFromSqs()
+        private async Task<int> GetNumberOfMessagesToReadFromSqsAsync()
         {
-            var numberOfMessagesToReadFromSqs = Math.Min(_messageProcessingStrategy.AvailableWorkers, MessageConstants.MaxAmazonMessageCap);
+            // TODO This needs reviewing as there can be available workers here now, yet by the time
+            // they're actually used, there could be none. This would then create a backlog of messages
+            // to process waiting on the message throttle because all the workers are busy.
+            int numberOfMessagesToReadFromSqs = Math.Min(_messageProcessingStrategy.AvailableWorkers, MessageConstants.MaxAmazonMessageCap);
```

Comment:
Any chance you can kill the `AvailableWorkers` property and always rely on `WaitForAvailableWorkerAsync` here. It could sidestep a bunch of bugs in any custom implementations.

Reply:
Done.